eos-update-flatpak-repos

```diff
@@ -635,9 +635,29 @@ def _migrate_installed_flatpaks():
             try:
                 new_branch = migrate.get('new-branch')
                 if new_branch:
-                    _update_branch(ref, new_branch, refs)
-                    branch = new_branch
-                    ref = _refresh_ref(inst, kind, name, arch, branch)
+                   new_ref = None
+                   try:
+                       new_ref = inst.get_installed_ref(kind, name, arch,
+                                                        new_branch)
+                   except GLib.GError as e:
+                       if e.matches(Flatpak.error_quark(),
+                                    Flatpak.Error.NOT_INSTALLED):
+                           pass
+                       else:
+                           raise
```

Comment:
```
if not e.matches(...):
    raise
```
?

Reply:
If you were feeling fancy you could write:

```python
try:
    new_ref = inst.get_installed_ref(...)
except GLib.GError as e:
    if not e.matches(...):
        raise
    # handle not-installed case
else:
    # handle already-installed case
```

but I'm not sure how popular `else` clauses are on `try` blocks.